src/plugins/OEmbed/Components.tsx

```diff
@@ -71,38 +78,42 @@ export const OEmbed = (
   if ("html" in props) {
     const processedHtml = preprocessHtml(props.html);
     return (
-      <Article
-        data-src={props.attributes.src}
-        data-spoilers={props.attributes.spoilers}
-        data-width={props.attributes.width}
-        data-height={props.attributes.height}
-        data-source={getWebsiteNameFromUrl(props.meta.canonical)}
-        data-loaded={props.loaded}
-        dangerouslySetInnerHTML={{ __html: processedHtml }}
-        ref={onAttachNode}
-      />
+      <BlockBaseComponent {...props} className="embed" enclosingTag={"article"}>
+        <div
+          className={wrapperClass}
+          dangerouslySetInnerHTML={{ __html: processedHtml }}
+          data-source={getWebsiteNameFromUrl(props.meta.canonical)}
+          data-loaded={props.loaded}
+          ref={onAttachNode}
+        />
+      </BlockBaseComponent>
     );
   }
   return <>Unimplemented</>;
 };
 
-export const OEmbedPlaceholder = (props: OEmbedData) => {
+// Not sure if we actually want the BlockBaseComponent here,
+// since if the width and height attributes have been set,
+// it will take up that space.
+// If the goal is to just have something that holds the data attributes,
+// it may be better to go back to a plain <article>
+export const OEmbedPlaceholder = (props: BlockBaseProps) => {
   return (
-    <article
-      data-src={props.src}
-      data-spoilers={props.spoilers}
-      data-width={props.width}
-      data-height={props.height}
+    <BlockBaseComponent
+      {...props}
+      className="embed-placeholder"
+      enclosingTag={"article"}
```

Comment:
FYI, I think most libraries these days standardize on this prop as `as={tag}`

Reply:
I found that confusing in the TipTap code. `as` is not a helpful prop name unless you are already familiar with that convention

Reply:
Agreed, I guess it's a matter of how much one should follow convention. I'm fine with this right now.